lib/tiappxml.js

```diff
@@ -817,9 +817,17 @@ function toJS(obj, doc) {
 					obj[node.tagName] = node.firstChild && node.firstChild.data.replace(/\n/g, '').trim() || '';
 					break;
 
+				case 'id':
+					if (xml.getAttr(node, 'platform') === targetPlatform || obj[node.tagName] === undefined) {
```

Comment:
By default, `targetPlatform` is an empty string. If you do `<id platform=""/>`, presumably this would be true.

Do you think `targetPlatform` should be initialized and defaulted to `undefined`? Would likely need to check `targetPlatform && xml.getAttr(node, 'platform') === targetPlatform`.

Reply:
good idea. I've pushed the change already but my test is still running. Will close this if its working